libraries/extractor/src/main/java/androidx/media3/extractor/metadata/id3/Id3Decoder.java

```diff
@@ -428,18 +429,24 @@ private static Id3Frame decodeFrame(
         String id = getFrameId(majorVersion, frameId0, frameId1, frameId2, frameId3);
         frame = decodeBinaryFrame(id3Data, frameSize, id);
       }
-      if (frame == null) {
-        Log.w(
-            TAG,
-            "Failed to decode frame: id="
-                + getFrameId(majorVersion, frameId0, frameId1, frameId2, frameId3)
-                + ", frameSize="
-                + frameSize);
-      }
-      return frame;
-    } finally {
+    } catch (Exception e) {
+      error = ",error=" + e.getMessage();
+    } catch (OutOfMemoryError e) {
```

Comment:
Can you combine both catch clauses together? (as `Exception | OutOfMemoryError`)
This would allow to move the `Log.w` line and `return null` inside the catch clause and make the code a lot cleaner.

Reply:
@tonihei The Log.w was split to help debug other issues, `decodeTxxxFrame` and others can return null and we have no log or information to help debug things when users complains about missing data.

If possible I'd like to keep it here. If you do not want the error field I can add 2 different log calls.

Reply:
I've kept the logging for `frame == null` but improved the logs to include the full exception stack trace.